cwms-data-api/src/main/java/cwms/cda/data/dao/timeseriesprofile/TimeSeriesProfileDao.java

```
@@ -0,0 +1,209 @@
+        // Initialize the where condition
+        Condition whereCondition = JooqDao.caseInsensitiveLikeRegexNullTrue(AV_TS_PROFILE.LOCATION_ID, locationIdMask);
+        whereCondition = whereCondition.and(JooqDao.caseInsensitiveLikeRegex(AV_TS_PROFILE.OFFICE_ID, officeIdMask));
+        whereCondition = whereCondition.and(JooqDao.caseInsensitiveLikeRegex(AV_TS_PROFILE.KEY_PARAMETER_ID,
+                parameterIdMask));
+
+        if (tsCursor != null) {
+            whereCondition = whereCondition.and(AV_TS_PROFILE.LOCATION_CODE.greaterThan(tsCursor));
+        }
+
+        // Get the total count if not already set
+        if (total == null) {
+            SelectConditionStep<Record1<Integer>> count = dsl.select(count(asterisk()))
+                .from(AV_TS_PROFILE)
+                .where(whereCondition);
+            total = count.fetchOne().value1();
+        }
+
+        // initialize the return object
+        TimeSeriesProfileList timeSeriesProfileList = new TimeSeriesProfileList.Builder()
+            .page(page)
+            .pageSize(pageSize)
+            .total(total)
+            .timeSeriesProfileList(new ArrayList<>())
+            .build();
+
+        // Get the time series profiles
+        @NotNull Result<Record> timeSeriesProfileResults;
+        SelectSeekStep2<Record, Long, String> selectionStep = dsl.select(DSL.asterisk()).from(AV_TS_PROFILE)
+            .where(whereCondition)
+            .orderBy(AV_TS_PROFILE.LOCATION_CODE, AV_TS_PROFILE.KEY_PARAMETER_ID);
+
+        // Use the cursor if it is set
+        // Start page at the cursor using JOOQ seek method
+        if (pageSize > 0) {
+            if (cursor != null) {
+                timeSeriesProfileResults = selectionStep.seek(tsCursor, parameterId).limit(pageSize + 1).fetch();
+            } else {
+                timeSeriesProfileResults = selectionStep.limit(pageSize + 1).fetch();
+            }
+        } else {
+            timeSeriesProfileResults = selectionStep.fetch();
```

I'd think a page size of <= 0 should throw an error.

---

Added an error throw in here. Previously treated negative numbers as override to get all results